Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManager.cpp

```diff
@@ -6,12 +6,99 @@
  *
  */
 
-#include <AzToolsFramework/ActionManager/Action/ActionManager.h>
+#include <QApplication>
+#include <QtGui/private/qguiapplication_p.h>
+#include <QShortcutEvent>
+
+#include <AzQtComponents/Components/WindowDecorationWrapper.h>
 
+#include <AzToolsFramework/ActionManager/Action/ActionManager.h>
 #include <AzToolsFramework/ActionManager/Menu/MenuManagerInterface.h>
 
 namespace AzToolsFramework
 {
+    //! This class is used to install an event filter on each widget that the action contexts are registered to in order to properly handle
+    //! ambiguous shorcuts.
+    class ActionContextWidgetWatcher : public QObject
+    {
+    public:
+        explicit ActionContextWidgetWatcher(QObject* parent)
+            : QObject(parent)
+        {
+        }
+
+        bool eventFilter(QObject* watched, QEvent* event) override
+        {
+            switch (event->type())
+            {
+            case QEvent::Shortcut:
+            {
+                QShortcutEvent* shortcutEvent = static_cast<QShortcutEvent*>(event);
+                if (shortcutEvent->isAmbiguous())
+                {
+                    QWidget* watchedWidget = qobject_cast<QWidget*>(watched);
+                    for (QAction* action : watchedWidget->actions())
+                    {
+                        if (action->shortcut() == shortcutEvent->key())
+                        {
+                            action->trigger();
+                            return true;
+                        }
+                    }
+                }
+                break;
+            }
+            case QEvent::ShortcutOverride:
+            {
+                QKeyEvent* keyEvent = static_cast<QKeyEvent*>(event);
+                int keyInt = keyEvent->key();
```

Comment:
suggestion/nit: Could this be called `keyIdentifier` perhaps or `keyCode` as opposed to `keyInt`?

Reply:
I like `keyCode` I'll go with that, thanks!